macros/src/derives/format/codegen/fields.rs

```diff
@@ -36,31 +36,33 @@ pub(crate) fn codegen(
             format_string.push_str(", ");
         }
 
+        let format_opt = get_defmt_format_option(field)?;
         let ty = as_native_type(&field.ty).unwrap_or_else(|| consts::TYPE_FORMAT.to_string());
-        if let Some(ident) = field.ident.as_ref() {
-            write!(format_string, "{}: {{={}:?}}", ident, ty).ok();
+        let ident = field
+            .ident
+            .clone()
+            .unwrap_or_else(|| format_ident!("arg{}", index));
+
+        if let Some(FormatOption::Debug2Format) = format_opt {
+            stmts.push(quote!(defmt::export::fmt(&defmt::Debug2Format(&#ident))));
+        } else if let Some(FormatOption::Display2Format) = format_opt {
```

Comment:
I suppose these didn't need to be `if let`s

Reply:
True, either normal if-else or match might be better suited.